File: Java/src/com/dbc/algorithm/ReferOffer/HeBingLiangGePaiXuDeLianBiaoLcof.java
package com.dbc.algorithm.ReferOffer;

public class HeBingLiangGePaiXuDeLianBiaoLcof {
    public class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        ListNode pre = new ListNode(0);
        ListNode temp = pre;
        while (l1 != null && l2 != null) {
            if (l1.val <= l2.val) {
                temp.next = l1;
                temp = temp.next;
                l1 = l1.next;
            } else {
                temp.next = l2;
                temp = temp.next;
                l2 = l2.next;
            }
        }
        if (l1 != null) {
            temp.next = l1;
        } else {
            temp.next = l2;
        }
        return pre.next;
    }
}

File: src/tmp/Python/refer-offer/he_bing_liang_ge_pai_xu_de_lian_biao_lcof.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def mergeTwoLists(l1: ListNode, l2: ListNode) -> ListNode:
    pre = ListNode(0)
    temp = pre
    while l1 and l2:
        if l1.val <= l2.val:
            temp.next = l1
            temp = temp.next
            l1 = l1.next
        else:
            temp.next = l2
            temp = temp.next
            l2 = l2.next
    
    temp.next = l1 if l1 else l2
    return pre.next
